iceoryx_hoofs/utility/include/iox/detail/unique_id.hpp

```diff
@@ -14,17 +14,15 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-#ifndef IOX_HOOFS_CXX_UNIQUE_ID_HPP
-#define IOX_HOOFS_CXX_UNIQUE_ID_HPP
+#ifndef IOX_HOOFS_UTILITY_UNIQUE_ID_HPP
+#define IOX_HOOFS_UTILITY_UNIQUE_ID_HPP
 
 #include "iox/newtype.hpp"
 
 #include <atomic>
 
 namespace iox
 {
-namespace cxx
```

Comment:
Could you please wrap the class in the `detail` namespace?

Reply:
@mossmaurice @elBoberido This class is used in posh. Shall we therefore make it public?

Reply:
Another question: Shall all classes in a detail folder be wrapped inside the `detail` namespace? If so, it's not consistent in hoofs, partly because not all internal classes were in the `internal` namespace before moving them.

Reply:
In this case I would say we follow the principle of least effort. The decision on what to put in `detail` or not was arbitrary anyway and therefore we can as well move it out of `detail`. I also don't think everything in `detail` should be in the `detail` namespace. Previously we didn't do this for `internal` either and the `*.inl` files in `detail` also do not have the `detail` namespace.

Reply:
Hmm.. I'm not too fond of our strategy here. However, I can see that those legacy things are sometimes not easy to clean-up. I can also see the benefit of keeping the namespace just `iox::`.

@FerdinandSpitzschnueffler Could you please adapt the sentence about `internal` in `iceoryx_hoofs/Readme.md` by replacing `s/internal/detail`. My feeling is that this will clear up some of things that can be misunderstood currently.

Reply:
@mossmaurice I think this needs a bit of discussion. Maybe in the next developer meetup. I wouldn't slow down the merge just because of this.

Reply:
Np, I'm ok with going ahead and merging. The next iceoryx developer meetup is already overloaded with too many topics, so I went ahead and created [an new issue](https://github.com/eclipse-iceoryx/iceoryx/issues/1921).